src/Library/demos/Color Picker/main.js

```diff
@@ -0,0 +1,26 @@
+import Gtk from "gi://Gtk";
+import Gdk from "gi://Gdk";
+import Gio from "gi://Gio";
+import GLib from "gi://GLib";
+import Xdp from "gi://Xdp";
+import XdpGtk from "gi://XdpGtk4";
+
+const portal = new Xdp.Portal();
+const parent = XdpGtk.parent_new_gtk(workbench.window);
+const button = workbench.builder.get_object("button");
+
+Gio._promisify(Xdp.Portal.prototype, "pick_color", "pick_color_finish");
+
+async function pickColor() {
+  try {
+    const color = await portal.pick_color(parent, null);
+    const colorInfo = color.print(false);
+
+    // The picked color is a GVariant of the form (ddd), containing red, green and blue components in the range [0,1].
+    console.log(`Selected Color is: ${colorInfo}`);
```

Comment:
wdyt we create a rgba color with https://docs.gtk.org/gdk4/struct.RGBA.html and then use `color.to_string()` ?

similar to the Color Dialog entry
You can read rgb from `result` using `const [r,g,b] = result.deepUnpack()`

Reply:
Sure